From 5f1101b566e060362cc92c80766981833f927607 Mon Sep 17 00:00:00 2001
From: Izak Burger <isburger@gmail.com>
Date: Mon, 13 Jan 2020 17:25:26 +0200
Subject: [PATCH 1/6] remove old Lynx Ion battery from battery settings

The old lynx ion caused DVCC to be forced on while the battery
does not send BOL information.

https://github.com/victronenergy/venus/issues/572
---
 delegates/batterysettings.py | 3 +--
 tests/batterysense_test.py   | 4 ++--
 2 files changed, 3 insertions(+), 4 deletions(-)

diff --git a/delegates/batterysettings.py b/delegates/batterysettings.py
index b0b47f1..4fb69dd 100644
--- a/delegates/batterysettings.py
+++ b/delegates/batterysettings.py
@@ -3,7 +3,6 @@ from delegates.dvcc import Dvcc
 
 # Battery IDs
 BATTERY_BYD = 0xB00A
-BATTERY_LYNX_ION = 0x0142
 BATTERY_DISCOVER_AES = 0xB016
 
 class BatterySettings(SystemCalcDelegate):
@@ -26,7 +25,7 @@ class BatterySettings(SystemCalcDelegate):
 
 		# Set good settings for known batteries. Force SVS off and DVCC on
 		# for some batteries.
-		if pid in (BATTERY_BYD, BATTERY_LYNX_ION, BATTERY_DISCOVER_AES):
+		if pid in (BATTERY_BYD, BATTERY_DISCOVER_AES):
 			self._settings['vsense'] = 2 # Forced Off
 			self._settings['tsense'] = 2 # Forced Off
 			self._settings['bol'] = 3 # Forced on
diff --git a/tests/batterysense_test.py b/tests/batterysense_test.py
index 1cd268c..c92d67d 100644
--- a/tests/batterysense_test.py
+++ b/tests/batterysense_test.py
@@ -655,8 +655,8 @@ class VoltageSenseTest(TestSystemCalcBase):
 		self._set_setting('/Settings/Services/Bol', 0)
 		self._set_setting('/Settings/SystemSetup/SharedVoltageSense', 1)
 
-		# BYD, Lynx BMS, Discover AES
-		for product_id in (0xB00A, 0x0142, 0xB016):
+		# BYD, Discover AES
+		for product_id in (0xB00A, 0xB016):
 			self._add_device('com.victronenergy.battery.ttyO2',
 				product_name='battery',
 				values={
-- 
2.17.1

